api/v1/storagecluster_types.go

```diff
@@ -247,14 +247,32 @@ type ManageCephBlockPools struct {
 }
 
 // ManageCephNonResilientPools defines how to reconcile ceph non-resilient pools
+// +kubebuilder:validation:XValidation:rule="!(self.useExistingOsds == true) || !has(self.resources)",message="resources should not be set when useExistingOsds is true"
```

Comment:
IMO we should only create CEL rules to prevent state that would cause problems, such as changing values that we don't support changing. To keep it simple, I don't think we should create rules for state that can safely be ignored. All of these rules seem unnecessarily complex to maintain.

Reply:
+1 its a lot complex, it took me a lot of tries to figure them out,
But they are working as expected now,

We can still make the call to have it or not, cc @Madhu-1

Reply:
i am fine for removing it, My thought processing is that CEL rules are for validation and it sets the right expectation for the users what need to be set and when and what is allowed and what is not allowed. if we put that in CRD comments no one will read it unless if the expectations are not meet.

Reply:
If we were discussing this before Parth made the changes may be I would have suggested to not add them. But if the changes are already in & working fine let's have them. Removing them would be just extra efforts.